File: src/main/java/tn/esprit/projet/models/ChatBotConfig.java
package tn.esprit.projet.models;

public class ChatBotConfig {
    private int id_config;
    private int id_psychologue;
    private int id_utilisateur;
    private String suggestion_personnalisee;
    private String frequence_rappel;

    // Constructeurs
    public ChatBotConfig() {}

    public ChatBotConfig(int id_psychologue, int id_utilisateur,
                         String suggestion_personnalisee, String frequence_rappel) {
        this.id_psychologue = id_psychologue;
        this.id_utilisateur = id_utilisateur;
        this.suggestion_personnalisee = suggestion_personnalisee;
        this.frequence_rappel = frequence_rappel;
    }

    // Getters normaux
    public int getid_config() {
        return id_config;
    }

    public int getid_psychologue() {
        return id_psychologue;
    }

    public int getid_utilisateur() {
        return id_utilisateur;
    }

    public String getsuggestion_personnalisee() {
        return suggestion_personnalisee;
    }

    public String getfrequence_rappel() {
        return frequence_rappel;
    }

    // Getters majuscules
    public int getId_config() {
        return id_config;
    }

    public int getId_psychologue() {
        return id_psychologue;
    }

    public int getId_utilisateur() {
        return id_utilisateur;
    }

    public String getSuggestion_personnalisee() {
        return suggestion_personnalisee;
    }

    public String getFrequence_rappel() {
        return frequence_rappel;
    }

    // Setters
    public void setid_config(int id_config) {
        this.id_config = id_config;
    }

    public void setid_psychologue(int id_psychologue) {
        this.id_psychologue = id_psychologue;
    }

    public void setid_utilisateur(int id_utilisateur) {
        this.id_utilisateur = id_utilisateur;
    }

    public void setsuggestion_personnalisee(String suggestion_personnalisee) {
        this.suggestion_personnalisee = suggestion_personnalisee;
    }

    public void setfrequence_rappel(String frequence_rappel) {
        this.frequence_rappel = frequence_rappel;
    }
}

File: src/Entity/ChatbotConfig.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;

use App\Repository\ChatbotConfigRepository;

#[ORM\Entity(repositoryClass: ChatbotConfigRepository::class)]
#[ORM\Table(name: 'chatbot_config')]
class ChatbotConfig
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id_config = null;

    public function getId_config(): ?int
    {
        return $this->id_config;
    }

    public function setId_config(int $id_config): self
    {
        $this->id_config = $id_config;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $id_psychologue = null;

    public function getId_psychologue(): ?int
    {
        return $this->id_psychologue;
    }

    public function setId_psychologue(int $id_psychologue): self
    {
        $this->id_psychologue = $id_psychologue;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $id_utilisateur = null;

    public function getId_utilisateur(): ?int
    {
        return $this->id_utilisateur;
    }

    public function setId_utilisateur(int $id_utilisateur): self
    {
        $this->id_utilisateur = $id_utilisateur;
        return $this;
    }

    #[ORM\Column(type: 'text', nullable: false)]
    private ?string $suggestion_personnalisee = null;

    public function getSuggestion_personnalisee(): ?string
    {
        return $this->suggestion_personnalisee;
    }

    public function setSuggestion_personnalisee(string $suggestion_personnalisee): self
    {
        $this->suggestion_personnalisee = $suggestion_personnalisee;
        return $this;
    }

    #[ORM\Column(type: 'integer', nullable: false)]
    private ?int $frequence_rappel = null;

    public function getFrequence_rappel(): ?int
    {
        return $this->frequence_rappel;
    }

    public function setFrequence_rappel(int $frequence_rappel): self
    {
        $this->frequence_rappel = $frequence_rappel;
        return $this;
    }

}
